src/Sylius/Component/Channel/Context/SingleChannel/ChannelContext.php

```diff
@@ -0,0 +1,49 @@
+<?php
+
+/*
+ * This file is part of the Sylius package.
+ *
+ * (c) Paweł Jędrzejewski
+ *
+ * For the full copyright and license information, please view the LICENSE
+ * file that was distributed with this source code.
+ */
+
+namespace Sylius\Component\Channel\Context\SingleChannel;
+
+use Sylius\Component\Channel\Context\ChannelContextInterface;
+use Sylius\Component\Channel\Context\ChannelNotFoundException;
+use Sylius\Component\Channel\Repository\ChannelRepositoryInterface;
+
+/**
+ * @author Kamil Kokot <kamil.kokot@lakion.com>
+ */
+final class ChannelContext implements ChannelContextInterface
```

Comment:
Not not renaming to `SingleChannelContext` and putting it in `Sylius\Component\Channel\Context` namespace? IMO it's overhead to create a separate namespace for that :)